pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -41,26 +45,91 @@ var log = logf.Log.WithName("DeploymentReconciler")
 type DeploymentReconciler struct {
-	client       kclient.Client
-	scheme       *runtime.Scheme
-	Deployment   *appsv1.Deployment
-	componentExt *v1beta1.ComponentExtensionSpec
+	client         kclient.Client
+	scheme         *runtime.Scheme
+	DeploymentList []*appsv1.Deployment
+	componentExt   *v1beta1.ComponentExtensionSpec
 }
 
 func NewDeploymentReconciler(client kclient.Client,
 	scheme *runtime.Scheme,
 	componentMeta metav1.ObjectMeta,
+	workerComponentMeta metav1.ObjectMeta,
 	componentExt *v1beta1.ComponentExtensionSpec,
-	podSpec *corev1.PodSpec) *DeploymentReconciler {
+	podSpec *corev1.PodSpec, workerPodSpec *corev1.PodSpec) *DeploymentReconciler {
 	return &DeploymentReconciler{
-		client:       client,
-		scheme:       scheme,
-		Deployment:   createRawDeployment(componentMeta, componentExt, podSpec),
-		componentExt: componentExt,
+		client:         client,
+		scheme:         scheme,
+		DeploymentList: createRawDeployment(componentMeta, workerComponentMeta, componentExt, podSpec, workerPodSpec),
+		componentExt:   componentExt,
 	}
 }
+func createRawDeployment(componentMeta metav1.ObjectMeta, workerComponentMeta metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec, //nolint:unparam
+	podSpec *corev1.PodSpec, workerPodSpec *corev1.PodSpec) []*appsv1.Deployment {
+	var deploymentList []*appsv1.Deployment
+	var workerNodeSize string
+	var pipelineParallelSize string
+	var tensorParallelSize string
+
+	isvcName := componentMeta.GetLabels()[constants.InferenceServiceLabel]
+	multiNodeEnabled := false
+
+	if workerComponentMeta.Name != "" {
+		multiNodeEnabled = true
+		workerNodeSize = componentMeta.GetAnnotations()[constants.WorkerNodeSize]
+		if parsedValue, err := strconv.Atoi(workerNodeSize); err == nil {
+			// Set pipelineParallelSize to workerNodeSize + 1 (head)
+			pipelineParallelSize = strconv.Itoa(parsedValue + 1)
+		} else {
+			log.Error(err, "Failed to convert pipelineParallelSize to int, using the WorkerSpec.Size)")
+		}
+	}
+
+	defaultDeployment := createRawDefaultDeployment(componentMeta, componentExt, podSpec)
+	if multiNodeEnabled {
+		// Use defaut value(1) if tensor-parallel-size is not set (gpu count)
+		tensorParallelSize = constants.DefaultTensorParallelSize
+
+		for _, container := range podSpec.Containers {
+			if container.Name == constants.InferenceServiceContainerName {
+				if value, exists := utils.GetEnvVarValue(container.Env, constants.TensorParallelSizeEnvName); exists {
+					// Use the environment variable value
+					tensorParallelSize = value
+				}
+				break
+			}
+		}
+		// Update GPU resource of default podSpec
+		addGPUResourceToDeployment(defaultDeployment, constants.InferenceServiceContainerName, tensorParallelSize)
+
+		// Set the environment variable for "isvc name" to the MODEL_NAME when multiNodeEnabled is true.
+		addEnvVarToDeploymentSpec(&defaultDeployment.Spec, constants.InferenceServiceContainerName, "MODEL_NAME", isvcName)
```

Comment:
Different runtimes specify things like model name, model path, or served model name in various ways. To accommodate this in KServe, when using multi-node setups, the model_name and model_dir are provided as environment variables so that each runtime can handle them according to its own requirements.